core/src/main/scala/org/apache/spark/sql/tispark/TiRowRDD.scala

```diff
@@ -41,6 +38,53 @@ class TiRowRDD(override val dagRequest: TiDAGRequest,
                @transient private val sparkSession: SparkSession)
     extends TiRDD(dagRequest, physicalId, tiConf, tableRef, session, sparkSession) {
 
+  protected val logger: Logger = log
+
+  override protected def getPartitions: Array[Partition] = {
+    val partitions = super.getPartitions
+
+    if (!dagRequest.getUseTiFlash)
+      return partitions
+
+    logger.info("testing tiflash logic")
```

Comment:
You should not assume this code path is for "testing" ONLY.

Reply:
done